lib/mail/parsers/rfc_2822.ex

```diff
@@ -44,7 +59,12 @@ defmodule Mail.Parsers.RFC2822 do
   Parses a RFC2822 timestamp to a DateTime with timezone
 
   [RFC2822 3.3 - Date and Time Specification](https://tools.ietf.org/html/rfc2822#section-3.3)
+
+  Also supports obsolete format described in [RFC2822 4.3](https://datatracker.ietf.org/doc/html/rfc2822#section-4.3)
+  and invalid timestamps encountered in the wild. The return value will be either a UTC DateTime, or an error tuple
+  returning the invalid date string.
```

Comment:
I'm very willing to take any feedback or omit this altogether, but I thought it'd make sense to document the change in return value. I'd considered either listing the formats supported or suggesting checking out `Mail.Parsers.RFC2822Test` to see which 'invalid formats' are supported, but wanted to get some feedback on whether or not y'all wanted specific documentation for non compliant date support.

Reply:
I think the test suite is sufficient for the supported formats. We should aim to be as permissive as any email client  so I'm sure there will be other instances of datetime formats we aren't supported that we need to adapt to